govc/vm/disk/change.go

```diff
@@ -0,0 +1,179 @@
+/*
+Copyright (c) 2014-2015 VMware, Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package disk
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"strings"
+
+	"github.com/vmware/govmomi/govc/cli"
+	"github.com/vmware/govmomi/govc/flags"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/units"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+type change struct {
+	*flags.VirtualMachineFlag
+
+	name     string
+	key      int
+	label    string
+	filePath string
+
+	bytes units.ByteSize
+	mode  string
+}
+
+func init() {
+	cli.Register("vm.disk.change", &change{})
+}
+
+func (cmd *change) Register(ctx context.Context, f *flag.FlagSet) {
+	cmd.VirtualMachineFlag, ctx = flags.NewVirtualMachineFlag(ctx)
+	cmd.VirtualMachineFlag.Register(ctx, f)
+	err := (&cmd.bytes).Set("0G")
+	if err != nil {
+		panic(err)
+	}
+	f.Var(&cmd.bytes, "size", "New disk size")
+	f.StringVar(&cmd.name, "disk.name", "", "Disk name")
+	f.StringVar(&cmd.label, "disk.label", "", "Disk label")
+	f.StringVar(&cmd.filePath, "disk.filePath", "", "Disk file name")
+	f.IntVar(&cmd.key, "disk.key", 0, "Disk unique key")
+	f.StringVar(&cmd.mode, "mode", "", fmt.Sprintf("Disk mode (%s)", strings.Join(vdmTypes, "|")))
+}
+
+func (cmd *change) Process(ctx context.Context) error {
+	if err := cmd.VirtualMachineFlag.Process(ctx); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (cmd *change) FindDisk(ctx context.Context, list object.VirtualDeviceList) (*types.VirtualDisk, error) {
+	var disks []*types.VirtualDisk
+	for _, device := range list {
+		switch md := device.(type) {
+		case *types.VirtualDisk:
+			switch {
```

Comment:
This looks the same as CheckDiskProperties?  If so, personally would prefer to see CheckDiskProperties return a bool and call it here to remove the duplicate code.